src/core/unit.h

```diff
@@ -460,6 +460,9 @@ typedef struct Unit {
         bool exported_log_ratelimit_interval:1;
         bool exported_log_ratelimit_burst:1;
 
+        /* Whether we've seen a pidfd during deserialization of this unit. */
+        bool deserialize_got_pidfd:1;
```

Comment:
Shouldn't this be reset back to false at the end of the deserialization? Or at the beginning, or both. We serialize on reload too no?

Reply:
It shouldn't matter, since if we're not switching between versions we know we're going to get a pidfd, but we can reset it sure.

Reply:
The field itself is not serialized, so should not be necessary?

Reply:
In `manager_clear_jobs_and_units` all `Unit` objects are flushed out.

Reply:
We serialize/deserialize not just on reexec, but in other cases too, where the object stays around

Reply:
yeah that's true - it can be dropped then, but imho it doesn't hurt and it's an extra safety

Reply:
Let's drop it as @YHNdnzj said, it shouldn't be necessary

Reply:
Nope. `manager_deserialize` -> `manager_deserialize_units` -> `manager_deserialize_one_unit` -> `unit_deserialize_state` is only called when `daemon-reload`/`daemon-reexec`.